docs: Update the disruption doc to v1beta1 (#4794)
aws/karpenter: website/content/en/preview/concepts/disruption.md

---

## Control Flow

Karpenter sets a Kubernetes [finalizer](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/) on each node it provisions.
The finalizer blocks deletion of the node object while the Termination Controller cordons and drains the node, before removing the underlying NodeClaim. Disruption is triggered by the Disruption Controller, by the user through manual disruption, or through an external system that sends a delete request to the node object.

### Disruption Controller

> Karpenter automatically discovers disruptable nodes and spins up replacements when needed. Karpenter disrupts nodes by executing one [automatic method](#methods) at a time, in order of Expiration, Drift, Emptiness, and then Consolidation. Each method varies slightly, but they all follow the standard disruption process:

Reviewer: I think this anchor link needs to be updated now. Can you change it to the other type of anchorlinking present elsewhere in this doc?
```suggestion
Karpenter automatically discovers disruptable nodes and spins up replacements when needed. Karpenter disrupts nodes by executing one [automatic method](#automatic_methods) at a time, in order of Expiration, Drift, Emptiness, and then Consolidation. Each method varies slightly, but they all follow the standard disruption process:
```

Reviewer: should also remove emptiness since it's not listed as a method anymore
```suggestion
Karpenter automatically discovers disruptable nodes and spins up replacements when needed. Karpenter disrupts nodes by executing one [automatic method](#methods) at a time, in order of Expiration, Drift, and then Consolidation. Each method varies slightly, but they all follow the standard disruption process:
```

Author: Nice catch!

1. Identify a list of prioritized candidates for the disruption method.

>    * If there are [pods that cannot be evicted](#pod-eviction) on the node, Karpenter will ignore the node and try disruption it later.

Reviewer: ```suggestion
   * If there are [pods that cannot be evicted](#pod-eviction) on the node, Karpenter will ignore the node and try disrupting it later.
```

Reviewer: or this one
```suggestion
   * If there are [pods that cannot be evicted](#pod-eviction) on the node, Karpenter will ignore the node and try disruption later.
```

    # Delete all nodes owned by any nodepool
    kubectl delete nodes -l karpenter.sh/nodepool

    # Delete all nodes owned by a specific nodepool
    kubectl delete nodes -l karpenter.sh/nodepool=$NODEPOOL_NAME
    ```
* **NodePool Deletion**: Nodes are owned by the NodePool through an [owner reference](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications) that launched them. Karpenter will gracefully terminate nodes through cascading deletion when the owning NodePool is deleted.

{{% alert title="Note" color="primary" %}}
By adding the finalizer, Karpenter improves the default Kubernetes process of node deletion.
When you run `kubectl delete node` on a node without a finalizer, the node is deleted without triggering the finalization logic. The instance will continue running in EC2, even though there is no longer a node object for it. The kubelet isn’t watching for its own existence, so if a node is deleted, the kubelet doesn’t terminate itself. All the pod objects get deleted by a garbage collection process later, because the pods’ node is gone.
{{% /alert %}}

## Automated Methods

> * **Expiration**: Karpenter will annotate nodes as expired and disrupt nodes after they have lived a set number of seconds, based on the NodePool's `spec.disruption.expireAfter` value. You can use node expiry to periodically recycle nodes due to security concerns.

Reviewer: should this reflect the fact that it's a status condition? Optionally you could say "mark" instead of annotate so that it doesn't hint towards a known field in the object meta
```suggestion
* **Expiration**: Karpenter will add a status condition for nodes as expired and disrupt nodes after they have lived a set number of seconds, based on the NodePool's `spec.disruption.expireAfter` value. You can use node expiry to periodically recycle nodes due to security concerns.
```

__Behavioral Fields__
- Weight                      
- Limits                      
- ConsolidationPolicy               
- ConsolidateAfter      
- ExpireAfter  
---      
##### EC2NodeClass
| Fields                        | One-way | Two-way |
|-------------------------------|:-------:|:-------:|
| Subnet Selector Terms         |         |    x    |
| Security Group Selector Terms |         |    x    |
| AMI Family                    |    x    |         |

> | AMI Selector     Terms        |         |    x    |

Reviewer: These look like odd spaces in the raw, is it trimmed down in the rendered version?

Author: Whoops, good catch 👍

### Node-Level Controls

> Nodes can be opted out of consolidation disruption by setting the annotation `karpenter.sh/do-not-consolidate: "true"` on the node.

Reviewer: This should be do-not-disrupt, no?
```suggestion
Nodes can be opted out of consolidation disruption by setting the annotation `karpenter.sh/do-not-disrupt: "true"` on the node.
```

Author: Good catch. This is what I get for trying to do a find and replace 😝

>     karpenter.sh/do-not-consolidate: "true"

Reviewer: Should change here too
